From e667be07faa17a029e319926aa6cfba71de6f7e6 Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Sun, 5 Oct 2025 12:23:24 +0200
Subject: [PATCH] fix vsx build under clang

This was seemingly accidentally dropped in:

https://github.com/opencv/opencv/commit/63ef786a3a0afcd44bf842f967656052d52dde09

The removed lines are in the clang path and now clang build
fails with missing vec_cvfo.
---
 modules/core/include/opencv2/core/vsx_utils.hpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/modules/core/include/opencv2/core/vsx_utils.hpp b/modules/core/include/opencv2/core/vsx_utils.hpp
index 4d5a694..8463642 100644
--- a/modules/core/include/opencv2/core/vsx_utils.hpp
+++ b/modules/core/include/opencv2/core/vsx_utils.hpp
@@ -399,6 +399,10 @@ VSX_REDIRECT_1RG(vec_ushort8, vec_ushort8, vec_popcntu, vec_popcnt)
 VSX_REDIRECT_1RG(vec_uint4,   vec_uint4,   vec_popcntu, vec_popcnt)
 VSX_REDIRECT_1RG(vec_udword2, vec_udword2, vec_popcntu, vec_popcnt)
 
+// converts between single and double precision
+VSX_REDIRECT_1RG(vec_float4,  vec_double2, vec_cvfo, __builtin_vsx_xvcvdpsp)
+VSX_REDIRECT_1RG(vec_double2, vec_float4,  vec_cvfo, __builtin_vsx_xvcvspdp)
+
 // converts word and doubleword to double-precision
 #ifdef vec_ctd
 #   undef vec_ctd
-- 
2.51.0

